### PR TITLE
e2e: schema 2 config test amendments

### PR DESCRIPTION
**Pull request by miminar**
- Quote process expansions.
- Use braces for variable expansions (e.g. ${var}).
- Wrap variable expandions with quotes.
- Added some comments.

A follow-up for #10805

### CODE REVIEW

**Comment by miminar** [comment]
[test]

**Comment by miminar** [comment]
[testextended][extended:core(images)][extended:core(builds)]

**Comment by miminar** [comment]
@stevekuznetsov @PTAL

**Comment by miminar** [comment]
Flake #10375 (job https://ci.openshift.redhat.com/jenkins/job/test_pr_origin_extended/460/).
And flake #10773 (job https://ci.openshift.redhat.com/jenkins/job/test_pr_origin/8717/).

File: test/end-to-end/core.sh [hunk 1]
```diff
@@ -248,10 +248,12 @@ os::cmd::expect_success 'oc policy add-role-to-user system:image-puller system:a
 os::cmd::try_until_text 'oc policy who-can get imagestreams/layers -n custom' 'system:anonymous'
 os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/custom/cross:namespace-pull"
 os::cmd::expect_success "docker pull ${DOCKER_REGISTRY}/custom/cross:namespace-pull-id"
-# unauthorized pushes return authorization errors, regardless of backing data
-os::cmd::expect_failure_and_text "docker push ${DOCKER_REGISTRY}/missing/image:tag"              "authentication required|not authorized"
-os::cmd::expect_failure_and_text "docker push ${DOCKER_REGISTRY}/custom/cross:namespace-pull"    "authentication required|not authorized"
-os::cmd::expect_failure_and_text "docker push ${DOCKER_REGISTRY}/custom/cross:namespace-pull-id" "authentication required|not authorized"
+# unauthorized pushes return authorization errors, regardless of backing data; error messages differ with each docker version
+#   docker 1.9:  not authorized
```
Comment (stevekuznetsov) [inline test/end-to-end/core.sh]: Still not convinced we want this

File: test/end-to-end/core.sh [hunk 2]
```diff
@@ -248,10 +248,12 @@ os::cmd::expect_success 'oc policy add-role-to-user system:image-puller system:a
...
+#   docker 1.10: authentication required
+os::cmd::expect_failure_and_text "docker push ${DOCKER_REGISTRY}/missing/image:tag"              "not authorized|authentication required"
```
Comment (stevekuznetsov) [inline test/end-to-end/core.sh]:
```sh
os::cmd::expect_failure_and_text "docker push '${DOCKER_REGISTRY}/missing/image:tag'"
``^

**Comment by stevekuznetsov** [comment]
Changes mostly LGTM, still not sure we want this in e2e core

**Comment by liggitt** [comment]
> still not sure we want this in e2e core

your wish is @miminar's follow up issue - https://github.com/openshift/origin/issues/10813

**Comment by stevekuznetsov** [comment]
@liggitt duly noted, thanks

File: test/end-to-end/core.sh (hunk 1 above)
Comment (miminar) [inline test/end-to-end/core.sh]: It's true that e2e test won't complete with docker 1.9. I've reverted it back.

**Comment by miminar** [comment]
Comments addressed.

**Comment by stevekuznetsov** [comment]
LGTM

**Comment by miminar** [comment]
@mfojtik merge?

**Comment by mfojtik** [comment]
@miminar still code freeze.

**Comment by miminar** [comment]
@mfojtik good to merge now?

**Comment by mfojtik** [comment]
[merge]

**Comment by miminar** [comment]
Flake #10663

**Pull request merged**